ghub.py

```diff
@@ -168,77 +172,65 @@ def __fallback_get_api_token():
     This is deprecated and exists only to maintain backward compatibility
     with older versions of ghub.
     """
-    token = git_cmd("config --get".split() + [GIT_CONFIG_TOKEN])
-    return token.decode('utf8')
+    return git_cmd("config --get".split() + [GIT_CONFIG_TOKEN])
 
 
 def get_branch():
     """Get the current local branch."""
-    branch_name = git_cmd("symbolic-ref HEAD".split()).decode('utf8')
+    branch_name = git_cmd("symbolic-ref HEAD".split())
     if branch_name:
         return branch_name.strip().replace("refs/heads/", "")
     else:
         print("ERROR: detached HEAD")
         raise SystemExit
 
 
-def get_user_and_repo(remote_name='origin', alt_name=None):
+def get_remote(name='origin', alt_name=None):
     """
     Retrieve the repo and user for the specified remote name.
 
     Typical values are 'upstream' and 'origin'. If alt_name is
     supplied, use it as an alternative repo name if remote_name is
     not found.
     """
-    def get_repo_line(cmd_output):
-        for line in cmd_output.splitlines():
-            line = line.strip()
-            if line.startswith(ORIGIN_LINE_START):
-                return line[len(ORIGIN_LINE_START):].strip().decode('utf8')
 
-    repo_line = get_repo_line(git_cmd(('remote show -n ' +
-                                       remote_name).split()))
-    if not repo_line:
-        print("This does not appear to be a git repository.")
-        raise SystemExit
-    if repo_line.find(':') < 1 and alt_name is not None:
-        repo_line = get_repo_line(git_cmd(('remote show -n ' +
-                                           alt_name).split()))
-    if repo_line.find(':') < 1:
+    url = git_cmd(('remote get-url ' + name).split())
+    if not url:
+        url = git_cmd(('remote get-url ' + alt_name).split())
+    if not url:
         print(("Unable to find remote repo named '%s'. Run:\n\t"
-               "git remote add %s ..." % (remote_name, remote_name)))
+               "git remote add %s ..." % (name, name)))
         raise SystemExit
     else:
-        _, user_name_repo = repo_line.split(':')
-        while user_name_repo.startswith('/'):
-            user_name_repo = user_name_repo[1:]
-        user_name, repo = user_name_repo.split('/')[-2:]
-        repo = repo.replace('.git', '')
-    return user_name, repo
+        githost, path = url.split(':', maxsplit=1)
+        host = githost.split('@', maxsplit=1)[-1]
+        username, repo = path.split('/', maxsplit=1)
+        if repo.endswith('.git'): repo = repo.replace('.git', '')
```

Comment:
I think that one liner is anti pattern in python. 
Probably best to 
```
if repo.endswith('.git'): 
   repo = repo.replace('.git', '')
```
or the pythonic one liner is `repo = repo.replace('.git', '') if repo.endswith('.git') else repo`

Reply:
Yeah, also `repo.replace('.git', '')` could be tightened to just `repo = repo[:-4]`. I'll fix both.